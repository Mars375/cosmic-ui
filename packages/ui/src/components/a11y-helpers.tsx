import * as React from 'react';

export const VisuallyHidden = ({ children }: { children: React.ReactNode }) => (
  <span className="absolute -m-px h-px w-px overflow-hidden p-0 [clip:rect(0,0,0,0)]">
    {children}
  </span>
);

export const SkipLink = ({
  href = '#main',
  children = 'Skip to content',
}: {
  href?: string;
  children?: React.ReactNode;
}) => (
  <a
    href={href}
    className="sr-only focus:not-sr-only focus:fixed focus:left-2 focus:top-2 focus:rounded-md focus:bg-cosmic-primary focus:px-3 focus:py-2 focus:text-cosmic-primaryForeground"
  >
    {children}
  </a>
);

VisuallyHidden.displayName = 'VisuallyHidden';
SkipLink.displayName = 'SkipLink';

