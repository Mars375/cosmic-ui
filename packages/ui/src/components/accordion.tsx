import * as React from 'react';
import * as RadixAccordion from '@radix-ui/react-accordion';
import { twMerge } from 'tailwind-merge';

export interface AccordionItem {
  value: string;
  header: React.ReactNode;
  content: React.ReactNode;
}

export interface AccordionProps {
  items: AccordionItem[];
  type?: 'single' | 'multiple';
  className?: string;
}

export const Accordion = ({ className, items, type = 'single' }: AccordionProps) => {
  return (
    <RadixAccordion.Root type={type as any} className={twMerge('w-full text-white', className)}>
      {items.map((it) => (
        <RadixAccordion.Item
          key={it.value}
          value={it.value}
          className="border-b border-cosmic-border"
        >
          <RadixAccordion.Header>
            <RadixAccordion.Trigger className="flex w-full items-center justify-between px-3 py-2 text-left text-sm hover:bg-white/5 focus-visible:ring-2 focus-visible:ring-cosmic-primary">
              <span>{it.header}</span>
              <span aria-hidden>â–¾</span>
            </RadixAccordion.Trigger>
          </RadixAccordion.Header>
          <RadixAccordion.Content className="px-3 py-3 text-sm text-white/80">
            {it.content}
          </RadixAccordion.Content>
        </RadixAccordion.Item>
      ))}
    </RadixAccordion.Root>
  );
};

Accordion.displayName = 'Accordion';
