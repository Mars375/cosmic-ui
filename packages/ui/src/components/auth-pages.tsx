import * as React from 'react';
import { Input } from './input';
import { Button } from './button';

export const LoginPage = () => {
  return (
    <div className="mx-auto max-w-sm rounded-md border border-cosmic-border bg-cosmic-surface p-6 text-white">
      <h1 className="mb-4 text-lg font-semibold">Sign in</h1>
      <form className="space-y-3">
        <div>
          <label className="mb-1 block text-sm">Email</label>
          <Input type="email" placeholder="you@company.com" />
        </div>
        <div>
          <label className="mb-1 block text-sm">Password</label>
          <Input type="password" placeholder="••••••••" />
        </div>
        <Button className="w-full">Sign in</Button>
      </form>
    </div>
  );
};

export const RegisterPage = () => {
  return (
    <div className="mx-auto max-w-sm rounded-md border border-cosmic-border bg-cosmic-surface p-6 text-white">
      <h1 className="mb-4 text-lg font-semibold">Create account</h1>
      <form className="space-y-3">
        <div>
          <label className="mb-1 block text-sm">Full name</label>
          <Input placeholder="Jane Doe" />
        </div>
        <div>
          <label className="mb-1 block text-sm">Email</label>
          <Input type="email" placeholder="you@company.com" />
        </div>
        <div>
          <label className="mb-1 block text-sm">Password</label>
          <Input type="password" placeholder="••••••••" />
        </div>
        <Button className="w-full">Create account</Button>
      </form>
    </div>
  );
};

LoginPage.displayName = 'LoginPage';
RegisterPage.displayName = 'RegisterPage';
