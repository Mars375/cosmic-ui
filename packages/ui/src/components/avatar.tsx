import * as React from 'react';
import * as RadixAvatar from '@radix-ui/react-avatar';
import { twMerge } from 'tailwind-merge';

export interface AvatarProps extends React.ComponentPropsWithoutRef<typeof RadixAvatar.Root> {
  src?: string;
  alt?: string;
  fallback?: string;
  size?: number | 'xs' | 'sm' | 'md' | 'lg' | 'xl';
  backgroundClassName?: string; // permet de contrÃ´ler le fond (light/dark)
}

export const Avatar = ({
  className,
  src,
  alt,
  fallback,
  size = 36,
  backgroundClassName = 'bg-muted',
  ...props
}: AvatarProps) => {
  const pixelSize = React.useMemo(() => {
    if (typeof size === 'number') return size;
    const map: Record<string, number> = { xs: 24, sm: 36, md: 48, lg: 64, xl: 80 };
    return map[size] ?? 36;
  }, [size]);

  return (
    <RadixAvatar.Root
      className={twMerge(
        'inline-flex items-center justify-center overflow-hidden rounded-full border',
        backgroundClassName,
        className,
      )}
      style={{ width: pixelSize, height: pixelSize }}
      {...props}
    >
      {src ? (
        <RadixAvatar.Image src={src} alt={alt} className="h-full w-full object-cover" />
      ) : null}
      <RadixAvatar.Fallback
        delayMs={300}
        className="h-full w-full flex items-center justify-center"
      >
        {fallback ?? 'ðŸ‘¤'}
      </RadixAvatar.Fallback>
    </RadixAvatar.Root>
  );
};

Avatar.displayName = 'Avatar';
