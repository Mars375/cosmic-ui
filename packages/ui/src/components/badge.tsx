import * as React from 'react';
import { twMerge } from 'tailwind-merge';

export type BadgeVariant = 'default' | 'secondary' | 'success' | 'warning' | 'destructive';

const styles: Record<BadgeVariant, string> = {
  default: 'bg-white/10 text-white',
  secondary: 'bg-cosmic-secondary text-black',
  success: 'bg-emerald-600 text-white',
  warning: 'bg-amber-600 text-black',
  destructive: 'bg-red-600 text-white',
};

export interface BadgeProps extends React.HTMLAttributes<HTMLSpanElement> {
  variant?: BadgeVariant;
}

export const Badge = ({ className, variant = 'default', ...props }: BadgeProps) => {
  return (
    <span
      className={twMerge(
        'inline-flex items-center rounded-md px-2 py-0.5 text-xs font-medium',
        styles[variant],
        className,
      )}
      {...props}
    />
  );
};
