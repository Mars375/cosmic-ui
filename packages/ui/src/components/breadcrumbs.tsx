import * as React from 'react';
import { twMerge } from 'tailwind-merge';

export interface Crumb {
  label: string;
  href?: string;
}

export interface BreadcrumbsProps extends React.HTMLAttributes<HTMLElement> {
  items: Crumb[];
  separator?: React.ReactNode;
}

export const Breadcrumbs = ({ className, items, separator = '/', ...props }: BreadcrumbsProps) => {
  return (
    <nav className={twMerge('text-sm text-white/80', className)} aria-label="Breadcrumb" {...props}>
      <ol className="inline-flex items-center gap-2">
        {items.map((item, idx) => (
          <li key={`${item.label}-${idx}`} className="inline-flex items-center gap-2">
            {item.href ? (
              <a href={item.href} className="hover:underline">
                {item.label}
              </a>
            ) : (
              <span aria-current="page" className="text-white">
                {item.label}
              </span>
            )}
            {idx < items.length - 1 && <span aria-hidden>{separator}</span>}
          </li>
        ))}
      </ol>
    </nav>
  );
};

Breadcrumbs.displayName = 'Breadcrumbs';
