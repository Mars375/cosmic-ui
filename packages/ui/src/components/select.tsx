import * as React from 'react';
import * as RadixSelect from '@radix-ui/react-select';
import { twMerge } from 'tailwind-merge';

const Select = RadixSelect.Root;

const SelectTrigger = React.forwardRef<
  React.ElementRef<typeof RadixSelect.Trigger>,
  React.ComponentPropsWithoutRef<typeof RadixSelect.Trigger>
>(({ className, children, ...props }, ref) => (
  <RadixSelect.Trigger
    ref={ref}
    className={twMerge(
      'flex h-10 w-full items-center justify-between rounded-md px-3 py-2 text-sm focus:outline-none disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 bg-background border border-input text-foreground placeholder:text-muted-foreground',
      className,
    )}
    {...props}
  >
    {children}
    <RadixSelect.Icon asChild>
      <svg
        width="16"
        height="16"
        viewBox="0 0 24 24"
        fill="none"
        xmlns="http://www.w3.org/2000/svg"
        className="h-4 w-4 opacity-50"
      >
        <path
          d="M6 9l6 6 6-6"
          stroke="currentColor"
          strokeWidth="2"
          strokeLinecap="round"
          strokeLinejoin="round"
        />
      </svg>
    </RadixSelect.Icon>
  </RadixSelect.Trigger>
));
SelectTrigger.displayName = RadixSelect.Trigger.displayName;

const SelectValue = RadixSelect.Value;

const SelectContent = React.forwardRef<
  React.ElementRef<typeof RadixSelect.Content>,
  React.ComponentPropsWithoutRef<typeof RadixSelect.Content>
>(({ className, children, position = 'popper', ...props }, ref) => (
  <RadixSelect.Portal>
    <RadixSelect.Content
      ref={ref}
      className={twMerge(
        'relative z-50 max-h-96 min-w-[8rem] overflow-hidden rounded-md shadow-md bg-popover border border-input text-foreground data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2',
        position === 'popper' &&
          'data-[side=bottom]:translate-y-1 data-[side=left]:-translate-x-1 data-[side=right]:translate-x-1 data-[side=top]:-translate-y-1',
        className,
      )}
      position={position}
      {...props}
    >
      <RadixSelect.Viewport
        className={twMerge(
          'p-1',
          position === 'popper' &&
            'h-[var(--radix-select-trigger-height)] w-full min-w-[var(--radix-select-trigger-width)]',
        )}
      >
        {children}
      </RadixSelect.Viewport>
    </RadixSelect.Content>
  </RadixSelect.Portal>
));
SelectContent.displayName = RadixSelect.Content.displayName;

const SelectItem = React.forwardRef<
  React.ElementRef<typeof RadixSelect.Item>,
  React.ComponentPropsWithoutRef<typeof RadixSelect.Item>
>(({ className, children, ...props }, ref) => (
  <RadixSelect.Item
    ref={ref}
    className={twMerge(
      'relative flex w-full cursor-default select-none items-center rounded-sm py-1.5 pl-8 pr-2 text-sm outline-none data-[disabled]:pointer-events-none data-[disabled]:opacity-50 hover:bg-muted',
      className,
    )}
    {...props}
  >
    <span className="absolute left-2 flex h-3.5 w-3.5 items-center justify-center">
      <RadixSelect.ItemIndicator>
        <svg
          width="14"
          height="14"
          viewBox="0 0 24 24"
          fill="none"
          xmlns="http://www.w3.org/2000/svg"
          className="h-4 w-4"
        >
          <path
            d="M20 6L9 17L4 12"
            stroke="currentColor"
            strokeWidth="2.5"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </RadixSelect.ItemIndicator>
    </span>

    <RadixSelect.ItemText>{children}</RadixSelect.ItemText>
  </RadixSelect.Item>
));
SelectItem.displayName = RadixSelect.Item.displayName;

export { Select, SelectContent, SelectItem, SelectTrigger, SelectValue };

