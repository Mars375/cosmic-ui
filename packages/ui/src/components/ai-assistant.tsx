'use client';

import * as React from 'react';
import { twMerge } from 'tailwind-merge';
import { Button } from './button';
import { Input } from './input';
import { Avatar } from './avatar';
import { Badge } from './badge';
import { Card, CardContent, CardHeader, CardTitle } from './card';

export interface AIMessage {
  id: string;
  content: string;
  sender: 'user' | 'ai';
  timestamp: Date;
  type?: 'text' | 'code' | 'suggestion' | 'error';
  suggestions?: string[];
  code?: {
    language: string;
    content: string;
  };
}

export interface AIAssistantProps {
  messages: AIMessage[];
  onSendMessage: (message: string) => void;
  onSuggestionClick?: (suggestion: string) => void;
  className?: string;
  placeholder?: string;
  isTyping?: boolean;
  disabled?: boolean;
  maxHeight?: number;
  showSuggestions?: boolean;
  suggestions?: string[];
}

export function AIAssistant({
  messages,
  onSendMessage,
  onSuggestionClick,
  className,
  placeholder = "Posez votre question à l'IA...",
  isTyping = false,
  disabled = false,
  maxHeight = 500,
  showSuggestions = true,
  suggestions = [],
}: AIAssistantProps) {
  const [inputValue, setInputValue] = React.useState('');
  const [isMinimized, setIsMinimized] = React.useState(false);
  const messagesEndRef = React.useRef<HTMLDivElement>(null);

  // Auto-scroll to bottom when new messages arrive
  React.useEffect(() => {
    messagesEndRef.current?.scrollIntoView({ behavior: 'smooth' });
  }, [messages]);

  const handleSendMessage = () => {
    if (inputValue.trim() && !disabled) {
      onSendMessage(inputValue.trim());
      setInputValue('');
    }
  };

  const handleKeyPress = (e: React.KeyboardEvent) => {
    if (e.key === 'Enter' && !e.shiftKey) {
      e.preventDefault();
      handleSendMessage();
    }
  };

  const handleSuggestionClick = (suggestion: string) => {
    if (onSuggestionClick) {
      onSuggestionClick(suggestion);
    } else {
      onSendMessage(suggestion);
    }
  };

  const formatTime = (timestamp: Date) => {
    return timestamp.toLocaleTimeString('fr-FR', {
      hour: '2-digit',
      minute: '2-digit',
    });
  };

  if (isMinimized) {
    return (
      <div className={twMerge('fixed bottom-4 left-4 z-50', className)}>
        <Button
          onClick={() => setIsMinimized(false)}
          className="rounded-full w-14 h-14 shadow-lg bg-cosmic-primary hover:bg-cosmic-primary/80"
        >
          <svg className="w-6 h-6" fill="none" stroke="currentColor" viewBox="0 0 24 24">
            <path
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth={2}
              d="M9.663 17h4.673M12 3v1m6.364 1.636l-.707.707M21 12h-1M4 12H3m3.343-5.657l-.707-.707m2.828 9.9a5 5 0 117.072 0l-.548.547A3.374 3.374 0 0014 18.469V19a2 2 0 11-4 0v-.531c0-.895-.356-1.754-.988-2.386l-.548-.547z"
            />
          </svg>
        </Button>
      </div>
    );
  }

  return (
    <div
      className={twMerge(
        'fixed bottom-4 left-4 w-96 bg-cosmic-surface border border-cosmic-border rounded-lg shadow-xl z-50',
        className,
      )}
    >
      {/* Header */}
      <div className="flex items-center justify-between p-4 border-b border-cosmic-border">
        <div className="flex items-center space-x-3">
          <div className="relative">
            <Avatar
              src="https://images.unsplash.com/photo-1472099645785-5658abf4ff4e?w=150&h=150&fit=crop&crop=face"
              alt="AI Assistant"
              size="sm"
            />
            <div className="absolute -bottom-1 -right-1 w-3 h-3 bg-green-400 rounded-full border-2 border-cosmic-surface" />
          </div>
          <div>
            <div className="font-medium text-white">Assistant IA</div>
            <div className="text-xs text-white/70">
              {isTyping ? "En train d'écrire..." : 'En ligne'}
            </div>
          </div>
        </div>
        <div className="flex items-center space-x-2">
          <Button
            variant="ghost"
            size="sm"
            onClick={() => setIsMinimized(true)}
            className="text-white/70 hover:text-white"
          >
            <svg className="w-4 h-4" fill="none" stroke="currentColor" viewBox="0 0 24 24">
              <path
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
                d="M19 9l-7 7-7-7"
              />
            </svg>
          </Button>
        </div>
      </div>

      {/* Messages */}
      <div className="overflow-y-auto p-4 space-y-4" style={{ maxHeight: `${maxHeight}px` }}>
        {messages.length === 0 ? (
          <div className="text-center text-white/50 py-8">
            <svg
              className="w-12 h-12 mx-auto mb-4 text-white/30"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
            >
              <path
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
                d="M9.663 17h4.673M12 3v1m6.364 1.636l-.707.707M21 12h-1M4 12H3m3.343-5.657l-.707-.707m2.828 9.9a5 5 0 117.072 0l-.548.547A3.374 3.374 0 0014 18.469V19a2 2 0 11-4 0v-.531c0-.895-.356-1.754-.988-2.386l-.548-.547z"
              />
            </svg>
            <p>Bonjour ! Comment puis-je vous aider ?</p>
          </div>
        ) : (
          messages.map((message) => (
            <div
              key={message.id}
              className={twMerge(
                'flex',
                message.sender === 'user' ? 'justify-end' : 'justify-start',
              )}
            >
              <div
                className={twMerge(
                  'flex max-w-[80%]',
                  message.sender === 'user' ? 'flex-row-reverse' : 'flex-row',
                )}
              >
                {message.sender === 'ai' && (
                  <Avatar
                    src="https://images.unsplash.com/photo-1472099645785-5658abf4ff4e?w=150&h=150&fit=crop&crop=face"
                    alt="AI Assistant"
                    size="xs"
                    className="mr-2 mt-1"
                  />
                )}

                <div
                  className={twMerge(
                    'rounded-lg px-3 py-2',
                    message.sender === 'user'
                      ? 'bg-cosmic-primary text-white'
                      : 'bg-cosmic-border text-white/90',
                  )}
                >
                  {message.type === 'code' && message.code ? (
                    <div className="space-y-2">
                      <div className="flex items-center justify-between">
                        <Badge variant="secondary" className="text-xs">
                          {message.code.language}
                        </Badge>
                        <Button
                          variant="ghost"
                          size="sm"
                          onClick={() => navigator.clipboard.writeText(message.code!.content)}
                          className="text-xs text-white/70 hover:text-white"
                        >
                          Copier
                        </Button>
                      </div>
                      <pre className="text-sm bg-cosmic-background rounded p-2 overflow-x-auto">
                        <code>{message.code.content}</code>
                      </pre>
                    </div>
                  ) : (
                    <div className="text-sm">{message.content}</div>
                  )}

                  {message.suggestions && message.suggestions.length > 0 && (
                    <div className="mt-3 space-y-2">
                      <div className="text-xs text-white/70">Suggestions :</div>
                      <div className="flex flex-wrap gap-2">
                        {message.suggestions.map((suggestion, index) => (
                          <Button
                            key={index}
                            variant="outline"
                            size="sm"
                            onClick={() => handleSuggestionClick(suggestion)}
                            className="text-xs"
                          >
                            {suggestion}
                          </Button>
                        ))}
                      </div>
                    </div>
                  )}

                  <div
                    className={twMerge(
                      'text-xs mt-1',
                      message.sender === 'user' ? 'text-white/70' : 'text-white/50',
                    )}
                  >
                    {formatTime(message.timestamp)}
                  </div>
                </div>
              </div>
            </div>
          ))
        )}

        {/* Typing indicator */}
        {isTyping && (
          <div className="flex justify-start">
            <div className="flex items-center space-x-2">
              <Avatar
                src="https://images.unsplash.com/photo-1472099645785-5658abf4ff4e?w=150&h=150&fit=crop&crop=face"
                alt="AI Assistant"
                size="xs"
              />
              <div className="bg-cosmic-border rounded-lg px-3 py-2">
                <div className="flex space-x-1">
                  <div className="w-2 h-2 bg-white/50 rounded-full animate-bounce" />
                  <div
                    className="w-2 h-2 bg-white/50 rounded-full animate-bounce"
                    style={{ animationDelay: '0.1s' }}
                  />
                  <div
                    className="w-2 h-2 bg-white/50 rounded-full animate-bounce"
                    style={{ animationDelay: '0.2s' }}
                  />
                </div>
              </div>
            </div>
          </div>
        )}

        <div ref={messagesEndRef} />
      </div>

      {/* Suggestions */}
      {showSuggestions && suggestions.length > 0 && messages.length === 0 && (
        <div className="p-4 border-t border-cosmic-border">
          <div className="text-xs text-white/70 mb-2">Suggestions :</div>
          <div className="flex flex-wrap gap-2">
            {suggestions.map((suggestion, index) => (
              <Button
                key={index}
                variant="outline"
                size="sm"
                onClick={() => handleSuggestionClick(suggestion)}
                className="text-xs"
              >
                {suggestion}
              </Button>
            ))}
          </div>
        </div>
      )}

      {/* Input */}
      <div className="p-4 border-t border-cosmic-border">
        <div className="flex items-center space-x-2">
          <Input
            value={inputValue}
            onChange={(e) => setInputValue(e.target.value)}
            onKeyPress={handleKeyPress}
            placeholder={placeholder}
            disabled={disabled}
            className="flex-1"
          />

          <Button onClick={handleSendMessage} disabled={!inputValue.trim() || disabled} size="sm">
            <svg className="w-4 h-4" fill="none" stroke="currentColor" viewBox="0 0 24 24">
              <path
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth={2}
                d="M12 19l9 2-9-18-9 18 9-2zm0 0v-8"
              />
            </svg>
          </Button>
        </div>
      </div>
    </div>
  );
}

// Hook for AI assistant
export function useAIAssistant() {
  const [messages, setMessages] = React.useState<AIMessage[]>([]);
  const [isTyping, setIsTyping] = React.useState(false);

  const addMessage = (message: Omit<AIMessage, 'id' | 'timestamp'>) => {
    const newMessage: AIMessage = {
      ...message,
      id: Math.random().toString(36).substr(2, 9),
      timestamp: new Date(),
    };
    setMessages((prev) => [...prev, newMessage]);
  };

  const sendMessage = (content: string) => {
    addMessage({
      content,
      sender: 'user',
    });
  };

  const simulateAIResponse = (
    content: string,
    type?: AIMessage['type'],
    suggestions?: string[],
    code?: AIMessage['code'],
  ) => {
    setIsTyping(true);
    setTimeout(
      () => {
        addMessage({
          content,
          sender: 'ai',
          type,
          suggestions,
          code,
        });
        setIsTyping(false);
      },
      1000 + Math.random() * 2000,
    );
  };

  const clearMessages = () => {
    setMessages([]);
  };

  return {
    messages,
    isTyping,
    addMessage,
    sendMessage,
    simulateAIResponse,
    clearMessages,
  };
}
