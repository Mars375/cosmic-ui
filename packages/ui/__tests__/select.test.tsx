import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Select } from '../src/components/select';
import { axe, toHaveNoViolations } from 'jest-axe';

expect.extend(toHaveNoViolations);

const items = [
  { value: 'a', label: 'Option A' },
  { value: 'b', label: 'Option B' },
];

test('opens and selects an item', async () => {
  const user = userEvent.setup();
  render(<Select items={items} ariaLabel="Plan" />);
  const trigger = screen.getByRole('combobox', { name: /plan/i });
  // Ouvrir via clavier (compatible jsdom)
  trigger.focus();
  await user.keyboard('{ArrowDown}');
  // Aller Ã  l'option B puis valider
  await user.keyboard('{ArrowDown}{Enter}');
  expect(trigger).toHaveTextContent(/option b/i);
});

test('is accessible', async () => {
  const { container } = render(<Select items={items} ariaLabel="Plan" />);
  const results = await axe(container);
  expect(results).toHaveNoViolations();
});

test('matches snapshot', () => {
  const { asFragment } = render(<Select items={items} ariaLabel="Plan" />);
  expect(asFragment()).toMatchSnapshot();
});
