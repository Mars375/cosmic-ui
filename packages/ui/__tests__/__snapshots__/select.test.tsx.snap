// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <button
    aria-autocomplete="none"
    aria-controls="radix-:ra:"
    aria-expanded="false"
    aria-label="Plan"
    class="inline-flex h-10 min-w-[10rem] items-center justify-between gap-2 rounded-lg border border-cosmic-border bg-cosmic-surface px-3 text-sm text-white outline-none transition-colors hover:bg-white/5 focus-visible:ring-2 focus-visible:ring-cosmic-primary"
    data-placeholder=""
    data-state="closed"
    dir="ltr"
    role="combobox"
    type="button"
  >
    <span
      style="pointer-events: none;"
    >
      Selectâ€¦
    </span>
    <span
      aria-hidden="true"
    >
      <svg
        fill="none"
        height="16"
        viewBox="0 0 24 24"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 9l6 6 6-6"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </span>
  </button>
</DocumentFragment>
`;
